### PR TITLE
rutabaga_gfx: fix build on Darwin

### PR DESCRIPTION
**Pull request by reckenrode**
## Description of changes

- Apply patch to gfxstream to remove the libdrm dependency on Darwin; and
- Fix rutabaga_gfx install phase on Darwin.

Found running nixpkgs-review for https://github.com/NixOS/nixpkgs/pull/298437.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by alyssais** [comment]
I was confused by this, because gfxstream is not supposed to depend on libdrm.  But it seems that the version we have packaged (which we can't upgrade right now because newer versions are [broken](https://lore.kernel.org/qemu-devel/wxhk5mcmsx2geak75py5ck6u5iis4vffmf7eekfjdyt65ol5qd@imbt42hgvgta/)) does unconditionally depend on DRM.  Perhaps we could just backport the [upstream patch](https://android.googlesource.com/platform/hardware/google/gfxstream/+/a8df2a3eb099b419a7b3638e68ea30b4cffb751b%5E%21/) that makes it optional?  It doesn't look too complicated…

**Comment by reckenrode** [comment]
@alyssais I applied the patch, but I also had to make a few other changes to get gfxstream to build on Darwin.

File: pkgs/development/libraries/rutabaga_gfx/default.nix
Comment (alyssais) [inline pkgs/development/libraries/rutabaga_gfx/default.nix:46]: libiconv should generally be depended on unconditionally.  On Linux it'll come from libc, but other platforms than Darwin need a separate libiconv.  (Academic for rutabaga_gfx, but it's a good principle in general.)

File: pkgs/development/libraries/rutabaga_gfx/default.nix
Comment (alyssais) [inline pkgs/development/libraries/rutabaga_gfx/default.nix:51]: Does this need to be reformatted?

**Review by alyssais — COMMENTED** [review]
Looks reasonable.  Do you feel like sending the patches upstream?

File: pkgs/development/libraries/rutabaga_gfx/default.nix
Comment (reckenrode) [reply pkgs/development/libraries/rutabaga_gfx/default.nix:51]: The whole file or just that block?

**Comment by reckenrode** [comment]
> Do you feel like sending the patches upstream?

I don’t know about the gfxstream change. It’s pretty nixpkgs specific. I looked at submitting the rutabaga_gfx change, but they require signing a CLA, which I’m generally disinclined to do.

**Comment by alyssais** [comment]
> I looked at submitting the rutabaga_gfx change, but they require signing a CLA, which I’m generally disinclined to do.

Good attitude.  That's fine by me.

File: pkgs/development/libraries/rutabaga_gfx/default.nix
Comment (alyssais) [reply pkgs/development/libraries/rutabaga_gfx/default.nix:51]: Well, any of it?

Comment (reckenrode) [reply pkgs/development/libraries/rutabaga_gfx/default.nix]: I guess that was kind of an odd question. I’ve been reformatting packages I maintain (or substantially change) with the RFC-166 formatter (nixpkgs-rfc-style) but holding off for other things. I went ahead and reformatted both. The force-push should have the changes.

File: pkgs/development/libraries/rutabaga_gfx/default.nix
Comment (reckenrode) [reply pkgs/development/libraries/rutabaga_gfx/default.nix]: Done. I made it unconditional.

File: pkgs/development/libraries/rutabaga_gfx/default.nix
Comment (alyssais) [reply pkgs/development/libraries/rutabaga_gfx/default.nix]: Ah, no, sorry — I meant, why did any of the formatting need to be changed, from the perspective that I'd rather have had the diff as small as possible than having any of the formatting changed, since formatting discussions are still ongoing.

Comment (reckenrode) [reply pkgs/development/libraries/rutabaga_gfx/default.nix]: Ah, okay. I’ll revert the formatting change and try to reduce the size further.

Comment (reckenrode) [reply pkgs/development/libraries/rutabaga_gfx/default.nix]: Done. I re-pushed with the formatting back the way it was and the formatting change to the `buildInputs` in rutabaga_gfx also reverted.

**Review by alyssais — APPROVED** [review]
Thanks a lot! Sorry for the miscommunication.

File: pkgs/development/libraries/gfxstream/default.nix
Comment (drupol) [inline pkgs/development/libraries/gfxstream/default.nix:33]:
```suggestion
    ++ lib.optionals stdenv.isDarwin [
```

File: pkgs/development/libraries/gfxstream/default.nix
Comment (drupol) [inline pkgs/development/libraries/gfxstream/default.nix:32]:
```suggestion
    ++ lib.optionals (lib.meta.availableOn stdenv.hostPlatform libdrm) [ libdrm ]
```

**Review by drupol — CHANGES_REQUESTED** [review]
Hello !

Don't you think we should use `optionals`?

It's true, it's opinionated. But `optional` vs `optionals` is always an issue for newcomers.

`optional` implicitly wrap the value in a list, while `optionals` doesn't. `optionals` have a lower cognitive load than `optional`, and I think that we should use it here.

File: pkgs/development/libraries/gfxstream/default.nix
Comment (reckenrode) [reply pkgs/development/libraries/gfxstream/default.nix:33]: That’s a typo, so I’ll fix this one. The intent was to use `optionals` with the `isDarwin` check.

File: pkgs/development/libraries/gfxstream/default.nix
Comment (reckenrode) [reply pkgs/development/libraries/gfxstream/default.nix:32]: Given that it’s checking whether libdrm is usable on the host platform, and the original derivation used `optional` not `optionals`, I’ll defer to @alyssais on whether to make the change here.

Comment (alyssais) [reply pkgs/development/libraries/gfxstream/default.nix:32]: I think either way is fine.

Comment (reckenrode) [reply pkgs/development/libraries/gfxstream/default.nix:32]: I’ll make the change for consistency with rutabaga_gfx, which uses `optionals`.

**Review by drupol — APPROVED** [review]

**Review by drupol — APPROVED** [review]

**Comment by reckenrode** [comment]
I pushed the change for `lib.optionals` and libdrm. Once I get an approval and the checks finish, I’ll merge into master.

**Review by alyssais — APPROVED** [review]

**Pull request merged**
